### DISCUSSION TITLE
Method for refreshing recipe metadata from original JSON/URL as Mealie feature partity with schema improves

### DISCUSSION THREAD

**Discussion by ponchohoncho** [mealie-recipes/mealie#1698] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
I love Mealie for keeping recipes from my favorite authors/chefs in one place without all the ads and mobile website weirdness. I don't typically edit the recipes once they're in Mealie. Mealie doesn't currently have have support for the author tag provided in the Recipe schema, making it difficult to search by author. It also only recently gained the ability to import tags and separate units/measurements. Recipes that were imported before these upgrades lack this new metadata. The only way to get it is to delete the recipe and reimport it. It would be great if there was a way to save the original JSON from the imported recipe so that it could be reprocessed to have more metadata extracted as feature parity with the schema continues to grow. If the original JSON is missing, then it can be retrieved from the original URL on Mealie upgrade or some other trigger.

**(Optional): Suggest A Solution**
The personal media server Plex has a feature to automatically download metadata about media in your library from an online source. If that data changes, or if Plex gains support for new fields, that metadata can be refreshed and whatever is downloaded can overwrite what's already there. Each field can be locked to prevent a refresh from erasing what you've edited though.

Mealie could work in a similar way. If a recipe is imported via URL, then the original JSON can be saved in the database alongside what Mealie parses out for its own internal structure and presentation. As support for things like tags, authors, split measurements, etc are built into Mealie, the JSON could be reprocessed and the recipe updated. That update would appear in the timeline noted in another request.

As long as the recipe is unedited, then all fields/areas are available to be refreshed. If a field is edited, like the title, or description, then that field is locked and will not be updated when the metadata is refreshed. This will keep customizations from being overwritten.

**Additional Information**
_FILL THESE OUT_
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by michael-genson** [comment]
Mealie's pretty stable at this point, so schema support is unlikely to change in a meaningful way to justify this feature.
